pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TransformFunctionFactory.java

```diff
@@ -112,13 +115,24 @@ public static TransformFunction get(TransformExpressionTree expression, Map<Stri
     switch (expression.getExpressionType()) {
       case FUNCTION:
         String functionName = expression.getValue();
-        Class<? extends TransformFunction> transformFunctionClass = TRANSFORM_FUNCTION_MAP.get(functionName);
+        Class<? extends TransformFunction> transformFunctionClass;
+        FunctionInfo functionInfo = null;
+        if (FunctionRegistry.containsFunctionByName(functionName)) {
+          transformFunctionClass = GenericTransformFunction.class;
+          functionInfo = FunctionRegistry.getFunctionByName(functionName);
```

Comment:
I am not sure I follow the logic here. Is GenericTransformFunction going to be the wrapper or single point of entry for all transform functions in Pinot. The name seems to suggest so but this code implies that only the functions registered in the registry can be treated as GenericTransformFunction. 

This also brings the point that we should add good javadocs to the new class.

Reply:
its a wrapper for all functions in FunctionRegistry. what name do you suggest - ScalarTransformFunctionWrapper